src/modules/swap/helpers/getSwapButtonState.ts

```diff
@@ -69,10 +69,14 @@ const quoteErrorToSwapButtonState: { [key in QuoteError]: SwapButtonState | null
 export function getSwapButtonState(input: SwapButtonStateParams): SwapButtonState {
   const { quoteError, approvalState } = input
 
+  // TODO
+  const isTokenSupportsEIP2612 = true
```

Comment:
My best guess, we will probably need a "list" of supported EIP-2612 tokens. AFAIU, there is not "general" way of supporting this because how the domain and `DOMAIN_SEPARATOR` are computed is under-specified in the EIP.

Reply:
Actually, there are lots of different implementations. However, we need to support only 5-6 main types